tests/snmp/test_snmp_phy_entity.py

```diff
@@ -634,31 +637,36 @@ def test_turn_off_psu_and_check_psu_info(duthosts, enum_supervisor_dut_hostname,
     pdu_controller = get_pdu_controller(duthost)
     if not pdu_controller:
         pytest.skip('psu_controller is None, skipping this test')
+
     outlet_status = pdu_controller.get_outlet_status()
     if len(outlet_status) < 2:
         pytest.skip(
             'At least 2 PSUs required for rest of the testing in this case')
 
-    # turn on all PSU
-    for outlet in outlet_status:
-        if not outlet['outlet_on']:
-            pdu_controller.turn_on_outlet(outlet)
-    time.sleep(5)
-
-    outlet_status = pdu_controller.get_outlet_status()
-    for outlet in outlet_status:
-        if not outlet['outlet_on']:
-            pytest.skip(
-                'Not all outlet are powered on, skip rest of the testing in this case')
-
-    # turn off the first PSU
-    first_outlet = outlet_status[0]
-    pdu_controller.turn_off_outlet(first_outlet)
-    assert wait_until(30, 5, 0, check_outlet_status,
-                      pdu_controller, first_outlet, False)
-    # wait for psud update the database
-    assert wait_until(180, 20, 5, _check_psu_status_after_power_off,
-                      duthost, localhost, creds_all_duts)
+    # Turn on all PDUs
+    logging.info("Turning all outlets on before test")
+    turn_on_all_outlets(pdu_controller)
+
+    PSU_to_PDUs = get_grouped_PDUs_by_PSU(pdu_controller)
+    try:
+        logging.info("Turning off PDUs connected to a random PSU")
+        # Get a random PSU's related PDUs to turn off
+        off_psu = random.choice(list(PSU_to_PDUs.keys()))
+        outlets = PSU_to_PDUs[off_psu]
+        logging.info("Toggling {} PDUs".format(off_psu))
+        for outlet in outlets:
+            pdu_controller.turn_off_outlet(outlet)
+            pytest_assert(wait_until(30, 5, 0, check_outlet_status,
+                          pdu_controller, outlet, False),
+                          "Outlet {} did not turn off".format(outlet['pdu_name']))
+
+        logging.info("Checking that turning off these outlets affects PSUs")
+        # wait for psud update the database
+        pytest_assert(wait_until(180, 20, 5, _check_psu_status_after_power_off,
```

Comment:
Let's discuss more on this wait timeout later as Cisco will need more time to wait for the SNMP result to come back.

Reply:
Set to 900s / 15m